### PR TITLE
enable metadata lock by default

### PR DESCRIPTION
**Pull request by Oreoxmt**
### What is changed, added or deleted? (Required)

Since TiDB v6.5.0, metadata lock is GA and enabled by default

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/12002
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by wjhuang2016 — APPROVED** [review]

File: sql-statements/sql-statement-commit.md [hunk 1]
```diff
@@ -39,7 +39,18 @@ Query OK, 0 rows affected (0.01 sec)
...
-* Currently, TiDB does not use Metadata Locking (MDL) to prevent DDL statements from modifying tables used by transactions by default. If the definition of a table has changed, committing a transaction will result in an `Information schema is changed` error. In this event, the transaction is rolled back automatically.
+<CustomContent platform="tidb">
+
+* Currently, TiDB use Metadata Locking (MDL) to prevent DDL statements from modifying tables used by transactions by default. The behavior of metadata lock is different between TiDB and MySQL. For more details, see [Metadata Lock](/metadata-lock.md).
+
+</CustomContent>
+
+<CustomContent platform="tidb-cloud">
+
+* Currently, TiDB use Metadata Locking (MDL) to prevent DDL statements from modifying tables used by transactions by default. The behavior of metadata lock is different between TiDB and MySQL. For more details, see [Metadata Lock](https://docs.pingcap.com/tidb/dev/metadata-lock).
```
Comment (TomShawn) [inline sql-statements/sql-statement-commit.md:50]: 是否确认下，元数据锁是否适用于 TiDB Cloud？一般来说，先发 OP 版本然后再上云，所以目前应该是不适用的吧？

**Review by TomShawn — APPROVED** [review]
rest LGTM

File: sql-statements/sql-statement-commit.md [hunk 2]
```diff
@@ -39,7 +39,18 @@ Query OK, 0 rows affected (0.01 sec)
...
+
+</CustomContent>
+
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-commit.md:53]:
```suggestion
```

**Pull request merged**